### ISSUE TITLE
broken with removing Coproduct from comonad

### ISSUE THREAD

**Issue by Heather** [ekmett/adjunctions#17]
related commit: https://github.com/ekmett/comonad/commit/f6070674f848034ca83699a78973062dac238b8b

problematic:

``` haskell
instance (Adjunction f g, Adjunction f' g') =>
         Adjunction (Coproduct f f') (Product g g') where
  unit a = Pair (leftAdjunct left a) (leftAdjunct right a)
  counit = coproduct (rightAdjunct fstP) (rightAdjunct sndP)
    where
      fstP (Pair x _) = x
      sndP (Pair _ x) = x
```

should be somehow replaced with Sum, not sure how

```
   * Variable not in scope: left :: f a -> Sum f f' a
    * Perhaps you meant one of these:
        data constructor `Left' (imported from Prelude),
        left' (imported from Data.Profunctor)
      Perhaps you want to add `left' to the import list in the import of
      `Control.Arrow' (src\Data\Functor\Adjunction.hs:37:1-35).
```

**Comment by RyanGlScott** [comment]
`Coproduct` was removed in favor of `Data.Functor.Sum`. It shouldn't be too hard to swap one for the other in `adjunctions`.

**Comment by Heather** [comment]
@RyanGlScott may you check question update and help with changing `Coproduct` to `Sum` in this instance?

**Comment by RyanGlScott** [comment]
Sure, I can make a pull request later today.

**Comment by Heather** [comment]
@RyanGlScott thank you

**Comment by RyanGlScott** [comment]
Fixed in 24ab91585504c0d237c8cd549aedccb438368ef6.
